src/musicbrainz/tagfetcher.cpp

```diff
@@ -299,9 +301,15 @@ void TagFetcher::slotMusicBrainzTaskSucceeded(
     auto pTrack = m_pTrack;
     terminate();
 
+    QString whyEmptyMessage;
+    if (guessedTrackReleases.empty()) {
+        whyEmptyMessage = tr("Could not find this track in the MusicBrainz database.");
+    }
+
     emit resultAvailable(
             std::move(pTrack),
-            std::move(guessedTrackReleases));
+            std::move(guessedTrackReleases),
+            whyEmptyMessage);
```

Comment:
the check is clunky and unnecessary from what I can tell. Can you just inline like you did with the other calls?

Reply:
so just
```suggestion
    emit resultAvailable(
            std::move(pTrack),
            std::move(guessedTrackReleases),
            tr("Could not find this track in the MusicBrainz database.");
```

Reply:
if unconditionally looking up the translated string presents too much overhead, we can also introduce an enum and switch on it only in the failure case (in `DlgTagFetcher::fetchTagFinished`).

Reply:
This way we add the text even if we have a list of tracks. I think it is clearer to have to dedicated emmit calls.

Reply:
yes, one in case of success and a different one in case of error would also be better.

Reply:
I rather thought of two different signals (with differing parameters), but the current solution works too ig. not widening the scope too much.